### PR TITLE
1337 pagination bar fix

### PR DESCRIPTION
**Pull request by ad9242**
## Summary of the changes
Fixes Pagination bar not updating when props change

## Related issue
#1337 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.

### CODE REVIEW

**Review by jd239 — CHANGES_REQUESTED** [review]
Looks good. A few formatting issues but functionality works as expected.

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx [hunk 1]
```diff
@@ -22,24 +23,30 @@ import {
   shadow: true,
 })
```
Comment (gd2910) [inline packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:24]:
```suggestion
})

```

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx [hunk 2]
```diff
@@ -22,24 +23,30 @@ import {
...
 export class PaginationBar {
-  private PAGINATION = "ic-pagination";
-  private TEXT_FIELD = "ic-text-field";
-  private TOOLTIP = "ic-tooltip";
   private PAGE_INPUT_FIELD_ID = "go-to-page-input";
```
Comment (gd2910) [inline packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:27]: Remove empty lines?

File: packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx [hunk 3]
```diff
@@ -69,11 +76,16 @@ export class PaginationBar {
   /**
    * The options which will be displayed for 'items per page' select input. Set a maximum of 4 options including a required 'All' option with value equal to total number of items.
    */
-  @Prop({ mutable: true }) itemsPerPageOptions?: {
+  @Prop() itemsPerPageOptions?: {
     label: string;
     value: string;
   }[];
```
Comment (gd2910) [inline packages/canary-web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:83]: Some props need to be alphabetised

**Review by gd2910 — APPROVED** [review]

**Comment by ad9242** [comment]
replaced by #1770

**Pull request closed without merging**